src/supervisor/watchers/index.ts

```diff
@@ -60,9 +60,18 @@ export async function beginWatchingWorkloads(): Promise<void> {
       { namespace: config.WATCH_NAMESPACE },
       'kubernetes-monitor restricted to specific namespace',
     );
-    const namespaceResponse = await k8sApi.coreClient.readNamespace(
-      config.WATCH_NAMESPACE,
-    );
+    let namespaceResponse: any;
+    try {
+      namespaceResponse = await k8sApi.coreClient.readNamespace(
+        config.WATCH_NAMESPACE,
+      );
+    } catch (err) {
+      logger.error(
+        { watchedNamespace: config.WATCH_NAMESPACE, err },
+        'failed to read the namespace',
+      );
+      return;
+    }
     const namespace = namespaceResponse.body;
```

Comment:
I think we can't assume the key `body` will exist?

Reply:
ie when `undefined`

Reply:
if there is an error, the `namespaceResponse` would be undefined, otherwise, It would have a `body`